client/state/data-layer/wpcom/sites/atomic/transfer/index.js

```diff
@@ -0,0 +1,163 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import {
+	ATOMIC_TRANSFER_INITIATE,
+	ATOMIC_TRANSFER_REQUEST,
+	ATOMIC_TRANSFERS_REQUEST,
+} from 'state/action-types';
+import { recordTracksEvent } from 'state/analytics/actions';
+import { dispatchRequestEx } from 'state/data-layer/wpcom-http/utils';
+import { requestSite } from 'state/sites/actions';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import {
+	fetchAtomicTransfer,
+	setAtomicTransfer,
+	atomicTransferComplete,
+} from 'state/atomic-transfer/actions';
+import { COMPLETED, ERROR } from 'state/atomic-transfer/constants';
+import { registerHandlers } from 'state/data-layer/handler-registry';
+import { pluginUploadError, updatePluginUploadProgress } from 'state/plugins/upload/actions';
+import { errorNotice } from 'state/notices/actions';
+
+/**
+ * Initiates a site transfer to Atomic.
+ *
+ * @param {?Object} action default action to call on HTTP events
+ * @returns {Object} action object
+ */
+export const initiateTransfer = action => {
+	const { siteId, module } = action;
+
+	return http(
+		{
+			apiNamespace: 'wpcom/v2',
+			method: 'POST',
+			path: `/sites/${ siteId }/atomic/transfers`,
+			query: {
+				plugin_zip: module.pluginZip,
+				plugin_slug: module.pluginSlug,
+				theme_zip: module.themeZip,
+			},
+		},
+		action
+	);
+};
+
+/**
+ * Retreives a list of Atomic transfer records.
+ *
+ * @param {?Object} action default action to call on HTTP events
+ * @returns {Object} action object
+ */
+export const getTransfers = action =>
```

Comment:
I'm not sure we need to fetch all the transfers for a given site. 
All we should be interested in is the latest transfer which the `sites/:site/transfers/latest` api endpoint should provide for us.
We should however consider moving that to `sites/:site/atomic/transfers/latest`

Reply:
This just corresponds to the default behavior for GET requests to that endpoint. 

> We should however consider moving that to sites/:site/atomic/transfers/latest

Good news, it's already there! 🎉
https://opengrok.a8c.com/source/xref/trunk/wp-content/rest-api-plugins/endpoints/sites-atomic-transfers.php#245

I'll add the corresponding data layer